offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -743,9 +864,143 @@ export class CkbHandler {
+    if (
+      !senderLockscript ||
+      (senderLockscript.codeHash === committeeMultisigLockscript.code_hash &&
+        senderLockscript.hashType === committeeMultisigLockscript.hash_type &&
+        senderLockscript.args === committeeMultisigLockscript.args)
+    ) {
+      logger.warn(`sender should not be committee multisig cell`);
+      return null;
+    }
+
+    if (tx.inputs.length > 1) {
+      for (let i = 1; i < tx.inputs.length; i++) {
+        const thisPreviousOutput = nonNullable(tx.inputs[i].previousOutput);
+        const thisPreHash = thisPreviousOutput.txHash;
+        const thisTxPrevious = await this.ckb.rpc.getTransaction(thisPreHash);
+        if (thisTxPrevious == null) {
+          return null;
+        }
+        const thisSenderLockscript = thisTxPrevious.transaction.outputs[Number(thisPreviousOutput.index)].lock;
+        if (
+          !thisSenderLockscript ||
+          thisSenderLockscript.codeHash !== senderLockscript.codeHash ||
+          thisSenderLockscript.hashType !== senderLockscript.hashType ||
+          thisSenderLockscript.args !== senderLockscript.args
+        ) {
+          logger.warn(`inputs contain different lockscripts`);
+          return null;
+        }
+      }
+    }
+
+    if (
+      recipientLockscript.codeHash !== committeeMultisigLockscript.code_hash ||
+      recipientLockscript.hashType !== committeeMultisigLockscript.hash_type ||
+      recipientLockscript.args !== committeeMultisigLockscript.args
+    ) {
+      logger.warn(`the first output must be committee multisig`);
+      return null;
+    }
+
+    if (tx.outputs.length > 1) {
+      for (let i = 1; i < tx.outputs.length; i++) {
+        const thisLockscript = nonNullable(tx.outputs[i].lock);
+        if (
+          thisLockscript.codeHash === committeeMultisigLockscript.code_hash &&
+          thisLockscript.hashType === committeeMultisigLockscript.hash_type &&
+          thisLockscript.args === committeeMultisigLockscript.args
+        ) {
+          logger.warn(`the outputs except first must not be committee multisig`);
+          return null;
+        }
+      }
+    }
+
+    let amount = BigInt(0);
+    let bridgeFee = BigInt(ForceBridgeCore.config.eth.lockNervosAssetFee);
+    let assetIdent = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    if (
+      recipientTypescript &&
+      recipientTypescript.codeHash === sudtTypescript.codeHash &&
+      recipientTypescript.hashType === sudtTypescript.hashType
+    ) {
+      amount = utils.readBigUInt128LE(tx.outputsData[0]);
+      assetIdent = utils.computeScriptHash({
+        code_hash: recipientTypescript.codeHash,
+        hash_type: recipientTypescript.hashType,
+        args: recipientTypescript.args,
+      });
+      bridgeFee = committeeMultisigCellCapacity;
+    } else if (!recipientTypescript) {
+      amount = committeeMultisigCellCapacity - bridgeFee;
```

Comment:
amount = committeeMultisigCellCapacity;
bridgeFee = 0;  // bridgeFee will be calculated from EthereumMintRecord later, bridgeFee = lockAmount - mintAmount

Reply:
bridge fee calculation logic refactored according to the design doc